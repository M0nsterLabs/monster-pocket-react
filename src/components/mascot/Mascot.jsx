import React from 'react';
import './Mascot.less';
export default class Mascot extends React.Component {
  static propTypes = {
    type : React.PropTypes.string,
    hide : React.PropTypes.bool
  };

  static defaultProps = {
    type : 'dro',
    hide : false
  };

  render () {
    const hideClass = (this.props.hide) ? 'mascot_hide' : '';
    return <div className={`mascot-wrapper mascot-wrapper_${this.props.type}`} ref={(mascot) => { this.domNode = mascot; }}>
      <div className={`mascot ${hideClass}`}>
        <div className={`mascot__face mascot__face_${this.props.type}`}></div>
      </div>
    </div>;
  }
}
