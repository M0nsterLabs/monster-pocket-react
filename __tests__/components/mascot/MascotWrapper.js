/* global describe: false, it: false, expect: false, beforeEach: false, afterEach: false, jest: false, test: false */
import React from 'react';
import ReactDOM from 'react-dom';
import {mount} from 'enzyme';
import Mascot from '../../../src/components/mascot/MascotWrapper.jsx';
let container;
let event;
const targetCoordsReturn = {
  top    : 100,
  left   : 100,
  width  : 100,
  height : 100
};
const notificationCoordsReturn = {
  top  : 120,
  left : 110
};
beforeEach(() => {
  jest.useFakeTimers();
  container = document.createElement('div');
  event = new Event('animationend');
});

afterEach(() => {
  ReactDOM.unmountComponentAtNode(container);
});

describe('Snapshots:', () => {
  test('free mascot wrapper', () => {
    const component = mount(<Mascot text="free mascot wrapper" timeout={0}/>);
    component.node.trigger.getTargetCoords = jest.fn(() => {
      return targetCoordsReturn;
    });
    component.node.trigger.calcNotificationCoords = jest.fn(() => {
      return notificationCoordsReturn;
    });
    const mascotNodeElement = ReactDOM.findDOMNode(component.node.trigger.target.domNode).getElementsByClassName('mascot')[0];
    mascotNodeElement.dispatchEvent(event);
    expect(component.html()).toMatchSnapshot();
  });

  test('free mascot wrapper with hide', () => {
    const component = mount(<Mascot text="free mascot wrapper with hide" timeout={0}/>);
    component.node.trigger.getTargetCoords = jest.fn(() => {
      return targetCoordsReturn;
    });
    component.node.trigger.calcNotificationCoords = jest.fn(() => {
      return notificationCoordsReturn;
    });

    component.setState({'hide': true});
    component.node.trigger.target.domNode.dispatchEvent(event);
    expect(component.node.status.umounted).toBe(true);
  });

  test('free mascot wrapper without hide', () => {
    const component = mount(<Mascot text="free mascot wrapper without hide" timeout={0}/>);
    component.node.trigger.getTargetCoords = jest.fn(() => {
      return targetCoordsReturn;
    });
    component.node.trigger.calcNotificationCoords = jest.fn(() => {
      return notificationCoordsReturn;
    });

    component.setState({'hide': false});
    component.node.trigger.target.domNode.dispatchEvent(event);
    expect(component.node.status.umounted).toBe(false);
  });

  test('free mascot wrapper with timer', async () => {
    const component = mount(<Mascot text="free mascot wrapper with timer" timeout={4000}/>);
    component.node.trigger.getTargetCoords = jest.fn(() => {
      return targetCoordsReturn;
    });
    component.node.trigger.calcNotificationCoords = jest.fn(() => {
      return notificationCoordsReturn;
    });
    const mascotNodeElement = ReactDOM.findDOMNode(component.node.trigger.target.domNode).getElementsByClassName('mascot')[0];
    const event = new Event('animationend');
    mascotNodeElement.dispatchEvent(event);
    jest.runAllTimers();
    expect(component.html()).toMatchSnapshot();
  });

  test('free mascot wrapper with closeMascot', async () => {
    const component = mount(<Mascot text="free mascot wrapper with closeMascot" timeout={0}/>, {attachTo: container});
    component.node.trigger.getTargetCoords = jest.fn(() => {
      return targetCoordsReturn;
    });
    component.node.trigger.calcNotificationCoords = jest.fn(() => {
      return notificationCoordsReturn;
    });
    component.instance().closeMascot();
    component.node.trigger.target.domNode.dispatchEvent(event);
    expect(component.html()).toMatchSnapshot();
  });
});
