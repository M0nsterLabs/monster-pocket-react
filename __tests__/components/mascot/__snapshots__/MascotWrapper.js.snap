exports[`Snapshots: free mascot wrapper 1`] = `"<div class=\"mascot-wrapper mascot-wrapper_dro\"><div class=\"mascot \"><div class=\"mascot__face mascot__face_dro\"></div></div></div>"`;

exports[`Snapshots: free mascot wrapper with closeMascot 1`] = `"<div class=\"mascot-wrapper mascot-wrapper_dro\"><div class=\"mascot mascot_hide\"><div class=\"mascot__face mascot__face_dro\"></div></div></div>"`;

exports[`Snapshots: free mascot wrapper with timer 1`] = `"<div class=\"mascot-wrapper mascot-wrapper_dro\"><div class=\"mascot mascot_hide\"><div class=\"mascot__face mascot__face_dro\"></div></div></div>"`;
